api/v1alpha3/suite_test.go

```diff
@@ -34,8 +34,13 @@ var (
 )
 
 func TestMain(m *testing.M) {
-	// Bootstrapping test environment
 	utilruntime.Must(AddToScheme(scheme.Scheme))
+
+	if os.Getenv("CAPI_DISABLE_TEST_ENV") != "" {
+		os.Exit(m.Run())
+	}
```

Comment:
Maybe it's time to put the generic TestMain code somewhere else let everything call it instead of updating code everywhere?

Reply:
Yeah maybe :). I'll take a closer look at how we can handle the few differences we have. But 95% is the same over and over again. (just with a bit more or less logging :))

Reply:
@vincepri I implemented a first version of a `RunTests` func in the envtest package as a basis for discussion, PTAL :)